boa_interner/src/lib.rs

```diff
@@ -180,178 +248,31 @@ impl Interner {
-
-    /// Symbol for the `"get"` string.
-    pub const GET: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(9)) };
-
-    /// Symbol for the `"set"` string.
-    pub const SET: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(10)) };
-
-    /// Symbol for the `"<main>"` string.
-    pub const MAIN: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(11)) };
-
-    /// Symbol for the `"raw"` string.
-    pub const RAW: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(12)) };
-
-    /// Symbol for the `"static"` string.
-    pub const STATIC: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(13)) };
-
-    /// Symbol for the `"prototype"` string.
-    pub const PROTOTYPE: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(14)) };
-
-    /// Symbol for the `"constructor"` string.
-    pub const CONSTRUCTOR: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(15)) };
-
-    /// Symbol for the `"implements"` string.
-    pub const IMPLEMENTS: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(16)) };
-
-    /// Symbol for the `"interface"` string.
-    pub const INTERFACE: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(17)) };
-
-    /// Symbol for the `"let"` string.
-    pub const LET: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(18)) };
-
-    /// Symbol for the `"package"` string.
-    pub const PACKAGE: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(19)) };
-
-    /// Symbol for the `"private"` string.
-    pub const PRIVATE: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(20)) };
-
-    /// Symbol for the `"protected"` string.
-    pub const PROTECTED: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(21)) };
-
-    /// Symbol for the `"public"` string.
-    pub const PUBLIC: Self = unsafe { Self::from_raw(NonZeroUsize::new_unchecked(22)) };
-
-    /// Creates a `Sym` from a raw `NonZeroUsize`.
-    const fn from_raw(value: NonZeroUsize) -> Self {
-        Self { value }
-    }
-
-    /// Retrieves the raw `NonZeroUsize` for this symbol.
-    const fn as_raw(self) -> NonZeroUsize {
-        self.value
-    }
-}
-
-impl Symbol for Sym {
-    #[inline]
-    fn try_from_usize(index: usize) -> Option<Self> {
-        index
-            .checked_add(Self::PADDING)
-            .and_then(NonZeroUsize::new)
-            .map(|value| Self { value })
-    }
-
-    #[inline]
-    fn to_usize(self) -> usize {
-        self.value.get() - Self::PADDING
+    /// Generates a new symbol for the provided [`str`] pointer.
+    ///
+    /// # Safety
+    ///
+    /// The caller must ensure `string` points to a valid
+    /// memory inside `head` and that it won't be invalidated
+    /// by allocations and deallocations.
+    unsafe fn generate_symbol(&mut self, string: InternedStr) -> Sym {
+        let next = Sym::new(self.len() + 1).expect("Adding one makes `self.len()` always `> 0`");
```

Comment:
In the `expect()` message, I would mention integer overflow, since it's the only situation when this can happen. Something like this:
```suggestion
        let next = Sym::new(self.len() + 1).expect("cannot generate interner symbol: integer overflow");
```